src/tests-bns/bns-integration-tests.ts

```diff
@@ -611,7 +609,7 @@ describe('BNS integration tests', () => {
       status: 'name-register',
       zonefile:
         '$ORIGIN zumrai.id\n$TTL 3600\n_http._tcp	IN	URI	10	1	"https://gaia.blockstack.org/hub/1EPno1VcdGx89ukN2we4iVpnFtkHzw8i5d/profile.json"\n\n',
-      zonefile_hash: '853cd126478237bc7392e65091f7ffa5a1556a33',
+      zonefile_hash: '0x853cd126478237bc7392e65091f7ffa5a1556a33',
```

Comment:
This should be reverted to use the original formatting (no `0x` prefix) as this is a `/v1` endpoint intended to be backwards compatible with the same endpoints from Stacks 1.0.

Reply:
The problem is that [handleBlockMessage](https://github.com/blockstack/stacks-blockchain-api/blob/master/src/event-stream/event-server.ts#L223) returns zonefile_hash without `0x`. 
<img width="449" alt="handleBlockMessage" src="https://user-images.githubusercontent.com/47666766/133407018-c83a6e0f-daac-465d-b2d7-58a466c6237b.png">

The [handleNewAttachmentMessage](https://github.com/blockstack/stacks-blockchain-api/blob/master/src/event-stream/event-server.ts#L557) returns zonefile_hash with `0x`. 
<img width="527" alt="handleNewAttachmentMessage" src="https://user-images.githubusercontent.com/47666766/133407074-39f565b3-4a11-4584-af37-872e00f2348a.png">


So it stores the information without `0x` in [updateNames](https://github.com/blockstack/stacks-blockchain-api/blob/master/src/datastore/postgres-store.ts#L5296-L5300) and updates it for zonefile with `0x` in [resolveBnsNames](https://github.com/blockstack/stacks-blockchain-api/blob/master/src/datastore/postgres-store.ts#L1602-L1604).

Initially, we were searching with tx_id, as the same table was used to store `zonefiles` and `tx_ids`. Now, we have separate tables for both. 

For now, what we can do is drop `0x` from the `handleNewAttachmentMessage` to search for the same `zonefile_hash` as inserted. What do you think about that?

Reply:
> For now, what we can do is drop 0x from the handleNewAttachmentMessage to search for the same zonefile_hash as inserted. What do you think about that?

That sounds fine. Normally we decode hex strings into buffers (`bytea` in postgres) because it's about 50% smaller in size. But my primary concern is ensuring these endpoints stay backwards compatible, regardless of how they are stored in the db.